### PR TITLE
doc: update Java interop document

### PR DESCRIPTION
**Pull request by utamori**
Pure Dart binding is now standard
This page needs to be updated so that newcomers are not confused

- Upgrade JNIGEN to v0.9.0
- Remove legacy C bindings

HosseinYousefi/jnigen_example#2

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.
- [x] This PR doesn't contain automatically generated corrections or text (Grammarly, LLMs, and similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn't use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

<details>
  <summary>Contribution guidelines:</summary><br>

  - See our [contributor guide](https://github.com/dart-lang/site-www/blob/main/CONTRIBUTING.md) for general expectations for PRs.
  - Larger or significant changes should be discussed in an issue before creating a PR.
  - Code changes should generally follow the [Dart style guide](https://dart.dev/effective-dart) and use `dart format`.
  - Updates to [code excerpts](https://github.com/dart-lang/site-shared/blob/main/doc/code-excerpts.md) indicated by `<?code-excerpt` need to be updated in their source `.dart` file as well.
</details>

### CODE REVIEW

File: src/content/interop/java-interop.md [hunk 1]
```diff
@@ -37,7 +37,6 @@ that uses `package:jnigen` to generate bindings for a simple class.
 
 - JDK
 - [Maven][]
-- (Optional) [`clang-format`][] to format the generated C bindings
 
 [Maven]: https://maven.apache.org/
 [`clang-format`]: https://clang.llvm.org/docs/ClangFormat.html
```
Comment (HosseinYousefi) [inline src/content/interop/java-interop.md:42]: This can also be removed.

**Review by HosseinYousefi — CHANGES_REQUESTED** [review]
There are other mentions of the C bindings still available in the text. Please also change them.

File: src/content/interop/java-interop.md (hunk 1 above)
Comment (utamori) [reply src/content/interop/java-interop.md]: done

**Review by HosseinYousefi — CHANGES_REQUESTED** [review]
I mentioned that there are other places where the C bindings are not removed.

However I cannot point you to them, [as GitHub doesn't let me comment on unchanged lines](https://github.com/isaacs/github/issues/1655).

It would be nice to [allow changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so we can commit to your fork.

So I'll try to review using links to the lines:

https://github.com/dart-lang/site-www/pull/5783/files#diff-ea7b67b16b35f54455c4e13a54b0e18a2645749b60ad4e960f94b288061e967fR88

```diff
-To generate the Dart (and C) bindings, run `jnigen` and
+To generate the Dart bindings, run `jnigen` and
```

---

https://github.com/dart-lang/site-www/pull/5783/files#diff-ea7b67b16b35f54455c4e13a54b0e18a2645749b60ad4e960f94b288061e967fL123

```diff
-you must build the dynamic libraries for `jni` and the generated C files. 
+you must build the dynamic library for `jni`. 
```

**Comment by utamori** [comment]
@HosseinYousefi 
I had missed it. Thanks for pointing that out.
If there's anything else, feel free to have it edited.

**Review by HosseinYousefi — APPROVED** [review]
LGTM! Thanks @utamori!

**Comment by HosseinYousefi** [comment]
@parlough Please merge whenever you see fit.

**Review by parlough — APPROVED** [review]
Thank you both!

**Pull request merged**